app/controllers/provider_interface/activity_log_controller.rb

```diff
@@ -1,11 +1,17 @@
 module ProviderInterface
   class ActivityLogController < ProviderInterfaceController
+    include Pagy::Backend
+
+    PAGY_PER_PAGE = 50
+
     def index
-      @events = GetActivityLogEvents.call(
-        application_choices: GetApplicationChoicesForProviders.call(
-          providers: current_provider_user.providers,
-        ),
-      ).page(params[:page] || 1).per(50)
+      application_choices = GetApplicationChoicesForProviders.call(
+        providers: current_provider_user.providers,
+      )
+      events = GetActivityLogEvents.call(
+        application_choices: application_choices,
+      )
```

Comment:
Just being pedantic here, but we don't need to repeat `application_choices`
```suggestion
      events = GetActivityLogEvents.call(application_choices:)
```